pkg/square/builder.go

```diff
@@ -410,10 +409,15 @@ func (e element) maxShareOffset() int {
 	return e.numShares + e.maxPadding
 }
 
-func worstCaseShareIndexes(blobs, maxSquareCapacity int) []uint32 {
+// worstCaseShareIndexes returns the largest possible share indexes for a set
+// of blobs at a given appversion. Largest possible is "worst" in that protobuf
+// uses varints to encode integers, so larger integers can require more bytes to
+// encode.
+func worstCaseShareIndexes(blobs int, appVersion uint64) []uint32 {
+	maxSquareSize := appconsts.SquareSizeUpperBound(appVersion)
```

Comment:
I don't think this would've been problematic because maxSquareCapacity was never going to be greater than the square size upper bound. I don't think it was possible to underestimate the square size